src/Backend/Teradata/ToFinalTable/SqlBuilder.php

```diff
@@ -219,8 +229,145 @@ public function getColumnsString(
+            TeradataQuote::quoteSingleIdentifier($destinationDefinition->getTableName())
+        );
+
+        foreach ($stagingTableDefinition->getColumnsNames() as $columnName) {
+            if (!$importOptions->isNullManipulationEnabled()) {
+                $columnsSet[] = sprintf(
+                    '%s = "src".%s',
+                    TeradataQuote::quoteSingleIdentifier($columnName),
+                    TeradataQuote::quoteSingleIdentifier($columnName),
+                );
+                continue;
+            }
+            if (in_array($columnName, $importOptions->getConvertEmptyValuesToNull(), true)) {
+                // values '' values from staging convert to NULL
+                $columnsSet[] = sprintf(
+                    '%s = CASE WHEN "src".%s = \'\' THEN NULL ELSE "src".%s END',
+                    TeradataQuote::quoteSingleIdentifier($columnName),
+                    TeradataQuote::quoteSingleIdentifier($columnName),
+                    TeradataQuote::quoteSingleIdentifier($columnName)
+                );
+            } else {
+                $columnsSet[] = sprintf(
+                    '%s = COALESCE("src".%s, \'\')',
+                    TeradataQuote::quoteSingleIdentifier($columnName),
+                    TeradataQuote::quoteSingleIdentifier($columnName)
+                );
+            }
+        }
+
+        if ($importOptions->useTimestamp()) {
+            $columnsSet[] = sprintf(
+                '%s = \'%s\'',
+                TeradataQuote::quoteSingleIdentifier(ToStageImporterInterface::TIMESTAMP_COLUMN_NAME),
+                $timestamp
+            );
+        }
+
+        if (!$importOptions->isNullManipulationEnabled()) {
+            $columnsComparisonSql = array_map(
+                static function ($columnName) use ($dest) {
+                    return sprintf(
+                        '%s.%s <> "src".%s',
+                        $dest,
+                        TeradataQuote::quoteSingleIdentifier($columnName),
+                        TeradataQuote::quoteSingleIdentifier($columnName)
+                    );
+                },
+                $stagingTableDefinition->getColumnsNames()
+            );
+        } else {
+            $columnsComparisonSql = array_map(
+                static function ($columnName) use ($dest) {
+                    return sprintf(
+                        'COALESCE(CAST(%s.%s AS VARCHAR(%s)), \'\') <> COALESCE("src".%s, \'\')',
+                        $dest,
+                        TeradataQuote::quoteSingleIdentifier($columnName),
+                        Teradata::DEFAULT_NON_LATIN_CHAR_LENGTH,
+                        TeradataQuote::quoteSingleIdentifier($columnName)
+                    );
+                },
+                $stagingTableDefinition->getColumnsNames()
+            );
+        }
+
+        return sprintf(
+            'UPDATE %s FROM (%s) "src" SET %s WHERE %s AND (%s)',
```

Comment:
lepsi odsazeni query (viz nize):
```suggestion
            <<<SQL
            UPDATE %s FROM (
              %s
            ) "src" 
            SET %s 
            WHERE %s 
              AND (%s)'
            SQL,
```